crates/oxc_linter/src/rules/eslint/no_plusplus.rs

```diff
@@ -78,7 +78,10 @@ declare_oxc_lint!(
     /// ```
     NoPlusplus,
     restriction,
-    pending
+    // This is "dangerous" because this will rewrite the code in a way that is not equivalent.
+    // For example, `++i` and `i++` will be rewritten as `i += 1` even though they are not the same.
+    // If the code depends on the order of evaluation, then this might break it.
+    dangerous_suggestion,
```

Comment:
```suggestion
    conditional_dangerous_suggestion,
```

Reply:
We could even make this `conditional_suggestion`, since this is unlikely to create semantically/syntactically invalid JavaScript.

Reply:
fair, there is part of me that fears this creating some really strange regressions, but I guess labeling it as a "suggestion" already implies it's not a guaranteed fix